pytorch_lightning/loggers/metrics_agg.py

```diff
@@ -0,0 +1,119 @@
+import functools
+import operator
+from typing import Dict, Callable, Mapping, Sequence
+
+MetricsT = Dict[str, float]
+MetricsAggFnT = Callable[[Sequence[MetricsT]], MetricsT]
+
+
+def merge_two_dicts(d1: Mapping, d2: Mapping, fn: Callable[[float, float], float], default_value: float = 0) -> Dict:
+    """Merges two dictionaries values with the given function.
```

Comment:
what is different metrics are aggregated differently max/mean

Reply:
As we discussed, the main, or at least the one which I can see purpose of such the aggregation is to log accumulation batches correctly. In such case, the most (singe?) natural way of aggregation is the averaging across batches (metrics for identical steps). So, it seems like one aggregation strategy might be enough.
If you see another options (not only the averaging of accum. steps), we can add dict of functions (one function for each metric name). And if it's None, use the default strategy for all metrics. What do you think?

Reply:
@justusschock pls ^^

Reply:
answered below, since this is not just an implementation detail, but a more fundamental decision on logging